### ISSUE TITLE
Set prop enabled to false on resize event

### ISSUE THREAD

**Issue by Dronax** [eddiemf/vue-affix#40]
Hello. How I can disable plugin on window resize event? I tried do:

    import Affix from 'vue-affix';

    var affix = Vue.use(Affix);

    const app = new Vue({
    el: '#app',

    data() {

    },

    mounted() {
       this.usePushStateForTabs('.spark-tabs');
       this.calcStickyWidth();
       this.$nextTick(function() {
          window.addEventListener('resize', this.calcStickyWidth);
       })
    },

    methods: {
     calcStickyWidth() {
        let width = $('.vue-affix').parent().width();

        $('.vue-affix').css('width', width + 'px');

        if(! this.isMobile) { affix.enabled = false; }
     },

     isMobile() {
        return $(window).width() > 768 ? true : false
     },
    }
    });

But my solution is not working. How I can disable plugin on window.resize? Thanks.

**Comment by eddiemf** [comment]
How exactly it's not working? 

I'm not sure if this `affix.enabled = false;` should work, it's been a long time I don't mess with Vue, but I think it would be better to define this in `data` and just pass it as a prop to the component.

Also the logic in your `isMobile()` function is wrong, although it's not what's making your code not work.

**Comment by Dronax** [comment]
@eddiemf what I need to do? Can you please give a example. I detect mobile device by screen size, if screen size > 768 then this is not mobile device.

**Comment by eddiemf** [comment]
Inside `data()` you'll have something like:

```
data() {
  affixIsEnabled: true,
}
```

And inside your event listener callback you'll set it to `false`/`true` depending on your condition.

Now you can use `this.affixIsEnabled` inside your template like this:

```
<affix class="menu" relative-element-selector="#example-content" style="width: 300px" :enabled="affixIsEnabled">
  <a href="#markup-1">Markup 1</a>
  <a href="#markup-2">Markup 2</a>
</affix>
```

This would be the correct flow of doing it.

Regarding your function logic, if the screen size is higher than 768 it means it's a desktop/tablet, not mobile, the logic is inverse.

**Comment by Dronax** [comment]
@eddiemf Nice) And then I can use on resize event this variable? Something like this:

     window.addEventListener('resize', function() {
         this.affixIsEnabled  = this.isMobile(); 
     });

**Comment by eddiemf** [comment]
Yep, but be careful with the logic, if you want to disable `vue-affix` on mobile devices you should do `this.affixIsEnabled = !this.ismobile()`, otherwise it would be enabled **only** in mobile.

**Comment by Dronax** [comment]
I founded a bug. When on resize I scroll to bottom end, vue-affix send my sidebar to up. Why?

**Comment by eddiemf** [comment]
I see that you're still loading the `0.5.0` version which has a bug that is fixed in `0.5.1`. Just update your version and it should be working fine.

**Comment by Dronax** [comment]
@eddiemf check please: https://jsfiddle.net/k2e1g80o/ scroll to bottom, and you Can see, that when you scrolling to end, sidebar is up

**Comment by eddiemf** [comment]
You're using `affixEnabled: false` as default, that's why it only works when you resize the window and this value gets updated :)

You should probably use a computed property instead of a method for these kind of computation, or just repeat the logic of `isMobile` to set the initial value of `affixEnabled` so that it can work properly when you start it on desktop or mobile.

**Comment by ExsyDev** [comment]
@eddiemf I tried `computed` property instead of `methods`, not working :( https://jsfiddle.net/k2e1g80o/1/

**Comment by eddiemf** [comment]
No, you didn't understand what I said. You're initializing `affixEnabled` always as `false`, so it will never be enabled when you initialize it, it will only work when you resize because you have a listener attached that will update the `affixEnabled` value.

Here's an example of how to use a computed property: https://jsfiddle.net/L3d5xryq/

**Comment by ExsyDev** [comment]
@eddiemf you 're removed `affixEnabled`. Why? How then I can disable plugin?

**Comment by eddiemf** [comment]
It was just an example, the plugin is automatically disabled based on the value of `isMobile()` because I'm using it as a computed property. 

All of these questions are completely not related with this plugin, I think it would be nice for you to take a deeper look at how Vue (and javascript too) works with data, because then you can apply this knowledge to fix any problem like this :)

**Comment by ExsyDev** [comment]
@eddiemf I do it, but does not matter is not working :( https://jsfiddle.net/k2e1g80o/2/ Why?

**Comment by eddiemf** [comment]
My example works the way you want, take a look at it and compare with yours. You're missing some basic points on Vue and about JavaScript which I cannot teach you, you'll have to learn them to understand it better.

**Comment by ExsyDev** [comment]
@eddiemf I tried you example and when I add `isMobile` to `affixEnabled` - not working, I have same bug. Please check it.
